MarkupsToModel/Logic/vtkSlicerMarkupsToModelLogic.h

```diff
@@ -53,30 +53,32 @@ class VTK_SLICER_MARKUPSTOMODEL_MODULE_LOGIC_EXPORT vtkSlicerMarkupsToModelLogic
   public vtkSlicerModuleLogic
 {
 public:
-
   static vtkSlicerMarkupsToModelLogic *New();
   vtkTypeMacro(vtkSlicerMarkupsToModelLogic, vtkSlicerModuleLogic);
   void PrintSelf(ostream& os, vtkIndent indent);
 
-  // Sets the markups node to be transformed
-  void SetMarkupsNode( vtkMRMLMarkupsFiducialNode* newMarkups, vtkMRMLMarkupsToModelNode* moduleNode );
-
   vtkSlicerMarkupsLogic* MarkupsLogic;
+
   // Updates the mouse selection type to create markups or to navigate the scene.
   void UpdateSelectionNode( vtkMRMLMarkupsToModelNode* markupsToModelModuleNode );
+
   // Updates closed surface or curve output model from markups
   void UpdateOutputModel( vtkMRMLMarkupsToModelNode* moduleNode );
 
   void ProcessMRMLNodesEvents( vtkObject* caller, unsigned long event, void* callData );
 
-  // Low-level interface for generating closed surface model without creating a parameter node
-  bool UpdateClosedSurfaceModel(vtkMRMLMarkupsFiducialNode* markupsNode, vtkMRMLModelNode* modelNode, bool smoothing = true, bool forceConvex = false, double delaunayAlpha = 0.0, bool cleanMarkups = true);
+  // DEPRECATED - Sets the input node to be processed
+  void SetMarkupsNode( vtkMRMLMarkupsFiducialNode* newMarkups, vtkMRMLMarkupsToModelNode* moduleNode );
+
+  // DEPRECATED
+  bool UpdateClosedSurfaceModel( vtkMRMLMarkupsFiducialNode* markupsNode, vtkMRMLModelNode* modelNode,
```

Comment:
Please don't deprecate UpdateClosedSurfaceModel and UpdateOutputCurveModel methods. We need these. See for example this application:
https://github.com/lassoan/SlicerSegmentEditorExtraEffects/blob/7b932ec64f49d82448e802bf7c6ef16079ad5b2f/SegmentEditorSurfaceCut/SegmentEditorSurfaceCutLib/SegmentEditorEffect.py#L377

Reply:
The replacement code in that case would be:

```controlPoints = vtk.vtkPoints()
slicer.vtkCreateModelUtil.vtkMRMLNodeToVtkPoints( inputMarkup, controlPoints )
polyData = vtk.vtkPolyData()
slicer.vtkCreateCurveUtil.GenerateLinearCurveModel( controlPoints, polyData )
outputModel.SetAndObservePolyData( polyData )```

So it's a bit longer than I originally forecasted. Wonder if it's worth simplifying. What do you think?

Reply:
(Actually, having thought a bit about that code, I just realized how strange "LinearCurve" is. I think I'll change it to GenerateLineModel...)

Reply:
It is expected that a module exposes its public API through its module logic and MRML classes. It is highly unusual to encourage developers to use a utility class directly.

vtkMRMLNodeToVtkPoints (name is not good, it should be just MRMLNodeToVtkPoints) functionality should be in the model and markup module (either in the MRML node or logic class).

It is not nice to pollute the namespace with a so class that has so generic name. At least the class name should start with the module name and instead of "util" it should be somewhat more specific (generator, converter, ...).

Having separate method for each interpolation type may be inconvenient for developers (e.g., to switch between different interpolator types, developers would need to add switch statements in their code).

In itself none of the above issues are significant, but overall it adds up to be an unusual and not very developer-friendly interface. Overall, while not ideal, I feel that simply not deprecating the existing interface in the logic would be a good solution. It would be the usual way to access module features and advanced developers could decide to use lower-level API if they prefer it. Maybe we simply overthink this, as in the long term things may be redesigned anyway.

Reply:
vtkMRMLNodeToVtkPoints => I don't understand your comment. What do you mean by "functionality should be in the model and markup module"? Which module(s) are you referring to?

Class names => I followed the convention of your earlier header files. It seems like we can have either one namespace with all our methods (which you changed, I assume because it was too many methods), or we can have a few different namespaces with fewer methods each (which has the issues you just described). Either way is non-ideal for a different reason. Please let me know how to proceed.

separate method for each interpolation type => I don't understand. In what use case would a developer be switching between these?

"simply not deprecating the existing interface in the logic would be a good solution" => Consider it done, then.